crates/wasm-mutate/src/lib.rs

```diff
@@ -280,46 +205,57 @@ impl<'wasm> WasmMutate<'wasm> {
-                SnipMutator,
-                CodemotionMutator,
-                FunctionBodyUnreachable,
-                AddCustomSectionMutator,
-                AddTypeMutator {
-                    max_params: 20,
-                    max_results: 20,
-                },
-                AddFunctionMutator,
-                RemoveSection::Custom,
-                RemoveSection::Empty,
-                ConstExpressionMutator::Global,
-                ConstExpressionMutator::ElementOffset,
-                ConstExpressionMutator::ElementFunc,
-                RemoveItemMutator(Item::Function),
-                RemoveItemMutator(Item::Global),
-                RemoveItemMutator(Item::Memory),
-                RemoveItemMutator(Item::Table),
-                RemoveItemMutator(Item::Type),
-                RemoveItemMutator(Item::Data),
-                RemoveItemMutator(Item::Element),
-                RemoveItemMutator(Item::Tag),
-                ModifyDataMutator {
-                    max_data_size: 10 << 20, // 10MB
-                },
-            )
-        );
+        const MUTATORS: &[&dyn Mutator] = &[
+            &PeepholeMutator::new(2),
+            &RemoveExportMutator,
+            &RenameExportMutator { max_name_size: 100 },
+            &SnipMutator,
+            &CodemotionMutator,
+            &FunctionBodyUnreachable,
+            &AddCustomSectionMutator,
+            &AddTypeMutator {
+                max_params: 20,
+                max_results: 20,
+            },
+            &AddFunctionMutator,
+            &RemoveSection::Custom,
+            &RemoveSection::Empty,
+            &ConstExpressionMutator::Global,
+            &ConstExpressionMutator::ElementOffset,
+            &ConstExpressionMutator::ElementFunc,
+            &RemoveItemMutator(Item::Function),
+            &RemoveItemMutator(Item::Global),
+            &RemoveItemMutator(Item::Memory),
+            &RemoveItemMutator(Item::Table),
+            &RemoveItemMutator(Item::Type),
+            &RemoveItemMutator(Item::Data),
+            &RemoveItemMutator(Item::Element),
+            &RemoveItemMutator(Item::Tag),
+            &ModifyDataMutator {
+                max_data_size: 10 << 20, // 10MB
+            },
+        ];
+
+        // Attempt all mutators, but start at an arbitrary index.
+        let start = self.rng().gen_range(0..MUTATORS.len());
+        for m in MUTATORS.iter().cycle().skip(start).take(MUTATORS.len()) {
+            let can_mutate = m.can_mutate(self);
+            log::trace!("Can `{}` mutate? {}", m.name(), can_mutate);
+            if !can_mutate {
+                continue;
+            }
+            log::debug!("attempting to mutate with `{}`", m.name());
+            match m.mutate(self) {
+                Ok(iter) => {
+                    log::debug!("mutator `{}` succeeded", m.name());
+                    return Ok(Box::new(iter.into_iter().map(|r| r.map(|m| m.finish()))));
```

Comment:
Interesting! I don't know if you remember, but this was what we originally attempted to write this as, but it was failing the borrow checker in like one of the last things that actually require polonious to fix. But seems not to require that anymore? Not sure what changed! Anyways, super happy to be rid of that nasty macro!!

Reply:
Hm good point! I suspect it is related to the by-val-`self` change which I actually just did as a refactoring so it would work with the dynamic dispatch array I had (and I noticed nothing actually needed the by-value). But hey if it works it works!